File: backend/src/main/kotlin/at/tieber/werwolf_web/api/mapper/NightStepMapper.java
package at.tieber.werwolf_web.api.mapper;

import at.tieber.werwolf_web.api.model.NightStepDto;
import at.tieber.werwolf_web.logic.NightStep;
import org.mapstruct.Mapper;

import java.util.List;

@Mapper(componentModel = "spring")
public interface NightStepMapper {

    List<NightStepDto> toDto(List<NightStep> nightSteps);

    NightStepDto toDto(NightStep nightStep);

}

File: backend/src/main/kotlin/at/tieber/werwolf_web/api/mapper/NightStepMapper.kt
package at.tieber.werwolf_web.api.mapper

import at.tieber.werwolf_web.api.model.NightStepDto
import at.tieber.werwolf_web.logic.NightStep
import org.mapstruct.Mapper

@Mapper(componentModel = "spring")
interface NightStepMapper {

    fun toDto(nightSteps: List<NightStep>): List<NightStepDto>

    fun toDto(nightStep: NightStep): NightStepDto

}
